src/transformStateWithClones.js

```diff
@@ -6,8 +7,62 @@
  *
  * @return {Object[]}
  */
+// function transformStateWithClones(state, actions) {
+//   // write code here
+//   const table = {
+//     ...state,
+//   };
+//   const result = [];
+
+//   for (const action of actions) {
+//     if (action.type === 'clear') {
+//       for (const delet in table) {
+//         delete table[delet];
+//       }
+//       console.log(table);
+//     } else if (action.type === 'removeProperties') {
+//       for (const remove of action.keysToRemove) {
+//         delete table[remove];
+//       }
+//     } else {
+//       Object.assign(table, action.extraData);
+//     }
+
+//     result.push({ ...table });
+//   }
+
+//   return result;
+// }
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyOfState = {
+    ...state,
+  };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(copyOfState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const remove of action.keysToRemove) {
+          delete copyOfState[remove];
+        }
+        break;
+
+      case 'clear':
+        for (const delet in copyOfState) {
+          delete copyOfState[delet];
```

Comment:
[CODE STYLE]: Instead of using a `for...in` loop and deleting each property individually, you can set `copyOfState` to an empty object `{}` to clear it when the action type is 'clear'. This is more efficient and cleaner.